pkgs/development/compilers/llvm/15/llvm/default.nix

```
@@ -329,6 +331,8 @@ in stdenv.mkDerivation (rec {
     # file and doesn't link zlib as well.
     # https://github.com/ClangBuiltLinux/tc-build/issues/150#issuecomment-845418812
     "-DLLVM_ENABLE_LIBXML2=OFF"
+  ] ++ optionals enableZstd [
+    "-DLLVM_PREFER_STATIC_ZSTD=${toString stdenv.hostPlatform.isStatic}"
```

As noted, the `LLVM_USE_STATIC_ZSTD` change was backported to LLVM 15: https://github.com/llvm/llvm-project/commit/4bd3f3759259548e159aeba5c76efb9a0864e6fa

```suggestion
    # We can use this option to make llvm to use static version of zstd.
    #
    # This option was backported to LLVM 15: https://github.com/llvm/llvm-project/commit/4bd3f3759259548e159aeba5c76efb9a0864e6fa
    "-DLLVM_USE_STATIC_ZSTD=${toString stdenv.hostPlatform.isStatic}"
```